Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -563,18 +559,19 @@ private void RegisterCommands(IWindsorContainer container)
             //note: convention: the registration name for commands is the type name, not the full type name.
             //Otherwise, namespaces would get in the way when binding to the menu items.
             RegisterCommandsWithPresenters(container);
+            
+            //var commandsForCommandMenuItems = Assembly.GetExecutingAssembly().GetTypes()
+            //    .Where(type => type.IsClass && typeof(ICommandMenuItem).IsAssignableFrom(type))
+            //    .Select(type => CommandNameFromCommandMenuName(type.Name))
+            //    .ToHashSet();
 
-            var commandsForCommandMenuItems = Assembly.GetExecutingAssembly().GetTypes()
-                .Where(type => type.IsClass && typeof(ICommandMenuItem).IsAssignableFrom(type))
-                .Select(type => CommandNameFromCommandMenuName(type.Name))
-                .ToHashSet();
-
-            container.Register(Classes.FromThisAssembly()
+            container.Register(Classes.FromAssemblyContaining<CommandBase>()
                 .Where(type => type.Namespace != null
                             && type.Namespace.StartsWith(typeof(CommandBase).Namespace ?? string.Empty)
                             && (type.BaseType == typeof(CommandBase) || type.BaseType == typeof(RefactorCommandBase))
                             && type.Name.EndsWith("Command")
-                            && commandsForCommandMenuItems.Contains(type.Name))
+                            // FIXME something something commandsForCommandMenuItems ...
```

Comment:
Hmm?

Reply:
the problem is that this might be a slightly different set of types than `commandsForCommandMenuItems` I'm not 100%  positive that the assumptions in that rewrite are correct

Reply:
We have quite a lot of implementations of `CommandBase ` that should not be registered in the IoC container. Removing the restriction causes problems when `IEnumerable<CommandBase>` gets injected.

Reply:
Two remarks: 

1) Putting all relevant names into a `HashSet` up front is quite probably faster than itterating all types for each command.
2) Please use the function `CommandNameFromCommandMenuName` to turn command menu item names into command names in a consistent manner.